### PR TITLE
refactor: Use manager client to get log for test

### PR DESCRIPTION
**Pull request by gaocegege**
Signed-off-by: Ce Gao <gaoce@caicloud.io>

**What this PR does / why we need it**:

Now the code to communicate with katib manager to get the log is hard coded in the main workflow. Thus we cannot run unit test because in unit test we do not have such a manager instance. This PR moves the function call to managerclient to mock it in test.

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/569)

### CODE REVIEW

File: pkg/controller/v1alpha2/trial/trial_controller_util.go [hunk 1]
```diff
@@ -96,29 +83,23 @@ func UpdateTrialStatusCondition(instance *trialsv1alpha2.Trial, deployedJob *uns
...
-	// read GetObservationLog call and update observation field
+func (r *ReconcileTrial) UpdateTrialStatusObservation(instance *trialsv1alpha2.Trial, deployedJob *unstructured.Unstructured) error {
 	objectiveMetricName := instance.Spec.Objective.ObjectiveMetricName
-	request := &api_pb.GetObservationLogRequest{
-		TrialName:  instance.Name,
-		MetricName: objectiveMetricName,
-	}
-	if reply, err := common.GetObservationLog(request); err != nil {
+	reply, err := r.GetTrialObservationLog(instance)
+	if err != nil {
```
Comment (johnugeorge) [inline pkg/controller/v1alpha2/trial/trial_controller_util.go]: Can you add a error og here? Missed earlier

Comment (gaocegege) [reply pkg/controller/v1alpha2/trial/trial_controller_util.go]: SGTM

**Comment by gaocegege** [comment]
@johnugeorge Addressed, PTAL

**Pull request merged**